### PR TITLE
PT-559: GOL iOS client as part of multiplatform project

### PR DESCRIPTION
**Pull request by tobiasheine**
Tracked in [PT-559](https://novoda.atlassian.net/browse/PT-559) 

### Description
This PR adds an `ios` module as part of our existing kotlin multiplatform project 🎉 
Since the integration into a multiplatform project is not yet officially released, we use a dev version of kotlin native compiler and gradle plugin.

Most of the swift source code has been copied from the previously added stand-alone [kotlin native client](https://github.com/novoda/spikes/tree/master/game-of-life-native). I will highlight the changes in the diff.

Besides that it fixes a bug in the setup of the `web` client which has been introduced on `master` by upgrading the gradle version.

### Considerations
- add the `iOS` client to the `README` 
- add `ios` platform module to existing project based on the stand-alone native client
- implement `actual GameLoopImpl` in kotlin using `NSTimer` 
- fix bug in `webpack` configuration after upgrade to gradle 4.4

### CODE REVIEW

File: game-of-life-multiplatform/build.gradle [hunk 1]
```diff
@@ -6,9 +6,12 @@ allprojects {
         repositories {
             google()
             jcenter()
+            maven { url "https://dl.bintray.com/jetbrains/kotlin-native-dependencies" }
         }
         dependencies {
             classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
+            classpath "org.jetbrains.kotlin:kotlin-native-gradle-plugin:+"
+            classpath "org.jetbrains.kotlin:kotlin-native-gradle-plugin:0.6-dev-936"
```
Comment (tobiasheine) [inline game-of-life-multiplatform/build.gradle]: We need to use a dev version of the `kotlin-native-gradle-plugin` since the feature has not yet been released officially.

File: game-of-life-multiplatform/gradle.properties [hunk 2]
```diff
@@ -16,4 +16,5 @@
 # http://www.gradle.org/docs/current/userguide/multi_project_builds.html#sec:decoupled_projects
 # org.gradle.parallel=true
 
-INCLUDE_ANDROID=true
\ No newline at end of file
+INCLUDE_ANDROID=true
+konan.version=0.6-dev-1005
```
Comment (tobiasheine) [inline game-of-life-multiplatform/gradle.properties]: This hardcodes the kotlin native compiler to a dev version which supports kotlin multiplatform projects. Basically it adds support for `expected` and `actual` declared kotlin classes.

File: game-of-life-multiplatform/iOS/src/main/kotlin/com/novoda/gol/GameLoop.kt [hunk 3]
```diff
@@ -0,0 +1,25 @@
+package com.novoda.gol
+
+import platform.Foundation.*
+
+actual class GameLoopImpl : GameLoop {
```
Comment (tobiasheine) [inline game-of-life-multiplatform/iOS/src/main/kotlin/com/novoda/gol/GameLoop.kt]: This has been changed compared to the native client where we implemented the [GameLoop](https://github.com/novoda/spikes/blob/master/game-of-life-native/Game%20of%20Life/SwiftGameLoop.swift) in swift.

In this case we implement it as `actual` kotlin class based on [NSTimer](https://developer.apple.com/documentation/foundation/nstimer).

File: game-of-life-multiplatform/build.gradle [hunk 4]
```diff
@@ -6,9 +6,12 @@ allprojects {
         repositories {
             google()
             jcenter()
+            maven { url "https://dl.bintray.com/jetbrains/kotlin-native-dependencies" }
         }
         dependencies {
             classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
+            classpath "org.jetbrains.kotlin:kotlin-native-gradle-plugin:+"
```
Comment (tasomaniac) [inline game-of-life-multiplatform/build.gradle]: this may not be necessary 👀

File: game-of-life-multiplatform/iOS/Game of Life/views/UICell.swift [hunk 5]
```diff
@@ -0,0 +1,26 @@
+//
```
Comment (tasomaniac) [inline game-of-life-multiplatform/iOS/Game of Life/views/UICell.swift]: Do we really need these headers?

File: game-of-life-multiplatform/settings.gradle [hunk 6]
```diff
@@ -1,3 +1,4 @@
 rootProject.name = 'game-of-life'
 
-include 'common', 'common-jvm', 'common-js', 'terminal-jvm', ':android', 'web', ':desktop-jvm'
+include 'common', 'common-jvm', 'common-js', 'terminal-jvm', ':android', 'web', ':desktop-jvm', 'iOS'
```
Comment (tasomaniac) [inline game-of-life-multiplatform/settings.gradle]: `nit` For consistency sake, I think we should have `:ios` here, also small case. What do you think?

File: game-of-life-multiplatform/README.md [hunk 7]
```diff
@@ -36,3 +36,12 @@ Run with `./gradlew desktop-jvm:run`
 
 ![gol](https://user-images.githubusercontent.com/517070/35706797-723ef2dc-07a7-11e8-857d-d4fc86a2e200.png)
 
+# iOS
+iOS client
+
+Requires xcode 9.2
```
Comment (devisnik) [inline game-of-life-multiplatform/README.md]: Where does this requirement come from?

File: game-of-life-multiplatform/web/webpack.common.js [hunk 8]
```diff
@@ -7,7 +7,7 @@ module.exports = {
         filename: "bundle.js"
     },
     resolve: {
-        modules: [path.resolve(__dirname, "node_modules"), path.resolve(__dirname, "build/classes/main/min/")]
+        modules: [path.resolve(__dirname, "node_modules"), path.resolve(__dirname, "build/kotlin-js-min/main/")]
```
Comment (devisnik) [inline game-of-life-multiplatform/web/webpack.common.js]: This is unrelated to the IOS client, right? Is it needed due to the plugin version update?

Comment (tasomaniac) [reply game-of-life-multiplatform/web/webpack.common.js]: Plugin version + Gradle version. The old JS setup we copied from kotlinconf app does not work with Gradle 4.+ and new native multi-platform does not work with older versions than Gradle 4.0.

Comment (tasomaniac) [reply game-of-life-multiplatform/web/webpack.common.js]: Here is the commit https://github.com/novoda/spikes/pull/366/commits/467c4521d374441bb20fe13476812b03181298c2

File: game-of-life-multiplatform/README.md (hunk 7 above)
Comment (tobiasheine) [reply game-of-life-multiplatform/README.md]: From [kotlin native](https://github.com/JetBrains/kotlin-native/blob/master/README.md)

Comment (tobiasheine) [reply game-of-life-multiplatform/README.md]: on macOS install Xcode 9.2
